cmd/agent/app/reporter/tchannel/reporter.go

```diff
@@ -106,3 +114,52 @@ func (r *Reporter) submitAndReport(submissionFunc func(ctx thrift.Context) error
 func (r *Reporter) CollectorServiceName() string {
 	return r.serviceName
 }
+
+func addAgentTagsToZipkinBatch(spans []*zipkincore.Span, agentTags []jaeger.Tag) []*zipkincore.Span {
+	for _, span := range spans {
+		for _, tag := range agentTags{
+			span.BinaryAnnotations = append(span.BinaryAnnotations, &zipkincore.BinaryAnnotation{
+				Key: tag.Key,
+				Value: []byte(*tag.VStr),
+				AnnotationType: 6, // static value set to string type for now.
+			})
+		}
+	}
+	return spans
+}
+
+func addAgentTags(spans []*jaeger.Span, agentTags []jaeger.Tag) []*jaeger.Span {
+	for _, span := range spans {
+		for _, tag := range agentTags{
+			span.Tags = append(span.Tags, &tag)
+		}
+	}
+	return spans
+}
+
+// Parsing logic borrowed from jaegertracing/jaeger-client-go
+func parseAgentTags(agentTags string) []jaeger.Tag {
```

Comment:
There is also `func parseAgentTags(agentTags string) []model.KeyValue {`.

Could we just have one version `func parseAgentTags(agentTags string) map[string][string]` or even return a list of tuples if we support multiple tags with the same keys and move it to flags.go.The map would be passed to reporter and at init time converted to list of appropriate model tags.

Reply:
Done.